syntax = 'proto3';

service Greetings {
    rpc SayHello (HelloRequest) returns (HelloReply) {}
    rpc DrawHello (DrawRequest) returns (DrawReply) {}
    rpc ImageHello (stream ImageRequest) returns (ImageReply) {}
    rpc VideoHello (stream VideoRequest) returns (stream VideoReply) {}
}

message HelloRequest {
    string name = 1;
}

message HelloReply {
    string message = 1;
}

message DrawRequest {
    string name = 1;
    bytes image = 2;
}

message DrawReply {
    string name = 1;
    bytes image = 2;
}

message ImageRequest {
    string name = 1;
    bytes image = 2;
}

message ImageReply {
    string name = 1;
}

message VideoRequest {
    string name = 1;
    bytes video = 2;
}

message VideoReply {
    string name = 1;
    bytes video = 2;
}
